src/components/PeoplePage.tsx

```diff
@@ -1,29 +1,74 @@
 import { PeopleFilters } from './PeopleFilters';
 import { Loader } from './Loader';
 import { PeopleTable } from './PeopleTable';
+import { useEffect, useState } from 'react';
+import { Person } from '../types';
+import { getPeople } from '../api';
+import { useSearchParams } from 'react-router-dom';
 
 export const PeoplePage = () => {
+  const [searchParam] = useSearchParams();
+  const sex = searchParam.get('sex') || '';
+  const nameSearch = searchParam.get('name') || '';
+  const centuries = searchParam.getAll('centuries') || [];
+  const sort = searchParam.get('sort') || '';
+  const direction = searchParam.get('direction') || '';
+
+  const [peopleData, setPeopleData] = useState<Person[] | null>(null);
+  const [loader, setLoader] = useState(false);
+  const [showError, setShowError] = useState(false);
+
+  useEffect(() => {
+    setLoader(true);
+    getPeople()
+      .then(db => {
+        setPeopleData(db);
+      })
+      .catch(() => {
+        setShowError(true);
+      })
+      .finally(() => {
+        setLoader(false);
+      });
+  }, []);
+
   return (
     <>
       <h1 className="title">People Page</h1>
 
       <div className="block">
         <div className="columns is-desktop is-flex-direction-row-reverse">
-          <div className="column is-7-tablet is-narrow-desktop">
-            <PeopleFilters />
-          </div>
+          {!loader && (
+            <div className="column is-7-tablet is-narrow-desktop">
+              <PeopleFilters centuries={centuries} sex={sex} />
+            </div>
+          )}
 
           <div className="column">
             <div className="box table-container">
-              <Loader />
-
-              <p data-cy="peopleLoadingError">Something went wrong</p>
+              {loader && <Loader />}
 
-              <p data-cy="noPeopleMessage">There are no people on the server</p>
+              {showError && (
+                <p data-cy="peopleLoadingError">Something went wrong</p>
+              )}
 
-              <p>There are no people matching the current search criteria</p>
+              {/* Show message if there are no people */}
+              {!loader && peopleData && peopleData.length === 0 && (
+                <p data-cy="noPeopleMessage">
+                  There are no people on the server
+                </p>
+              )}
 
-              <PeopleTable />
+              {!loader && peopleData && peopleData.length > 0 && (
+                <PeopleTable
+                  sort={sort}
+                  direction={direction}
+                  centuries={centuries}
+                  peopleData={peopleData}
+                  sex={sex}
+                  nameSearch={nameSearch}
+                />
```

Comment:
The sorting functionality is missing. Implement sorting logic in the `PeopleTable` component using the `sort` and `direction` parameters to sort the data by `name`, `sex`, `born`, and `died` as required.